File: src/python/src/de/yaxgl/Container/WindowManager.py
from xml.dom.minidom import parse
import wx
from de.yaxgl.Helper.LookAndFeel import LookAndFeel
from de.yaxgl.Container.Window import Window



class NotYetInitializedError(Exception):
    pass

class WindowManager:
    instance = None
    isInitialized = False
    
    def __init__(self):
        self.windows = []
        self.baseWindow = None
        self.app = wx.App(0)

    @classmethod
    def initialize(cls, lookAndFeel):
        if(lookAndFeel != LookAndFeel.System):
            raise NotSupportedException("Error: The LookAndFeel." + str(lookAndFeel) + " style is not supported in Python/wxWidgets applications.")
        else:
            cls.isInitialized = True

    @classmethod
    def getInstance(cls):
        if(not cls.isInitialized):
            raise NotYetInitializedException("Error: You have to initialize the WindowManager with a LookAndFeel before you can get an instance");

        elif(not cls.instance):
            cls.instance = WindowManager()
            
        return(cls.instance)

    def createWindow(self, xmlFile, eventReceiver):
        window = Window(xmlFile)
        window.registerEventHandlers(eventReceiver)
        self.windows.append(window)
        return(window)

    def run(self, window):
        #if(baseWindow == None):
        #    raise
        #Application.Run((Form)baseWindow.getNativeComponent());
        window.show()
        self.app.MainLoop()

    def show(self, window):
        window.Show(True)

    def showModal(self, window, parentWindow):
        pass

    def hide(self, window):
        window.Show(False)

    def close(self, window):
        if(window in self.windows):
            del self.windows[window]
        window.Close()

    def getWindowsByID(self, ID):
#===============================================================================
#            IList<Window> matchingWindows = new List<Window>();
#            foreach (Window window in windows)
#            {
#                if (window.getID().Equals(ID))
#                    matchingWindows.Add(window);
#            
#            }
#            return matchingWindows;
#===============================================================================
        pass

File: src/gwt/yaxgl-gwt-lib/src/de/yaxgl/client/Container/WindowManager.java
package de.yaxgl.client.Container;

import java.util.ArrayList;
import java.util.List;


import com.google.gwt.user.client.ui.RootPanel;


public class WindowManager {


	 List windows = null;
     private static WindowManager instance = null;
     private static boolean initialized = false;
     private static RootPanel display = null;
     
     private WindowManager()
     {
         windows = new ArrayList();
     }


     public static void initialize()
     {
         initialized = true;
     }
     
     
     public static WindowManager getInstance() throws Exception
     {
         if (!initialized)
            //TODO //throw new NotYetInitilaizedException("Error: You have to initialize the WindowManager with a LookAndFeel before you can get an instance");
        	 throw new Exception("The WindowManager must be initialized before you can have an instance.");
         
         if (instance == null)
         {
       		 display = RootPanel.get();
        	 instance = new WindowManager();
         }
         return instance;
     }
     
     public Window createWindow(String xmlfile)
     {
         Window window = new Window(display,xmlfile);
         windows.add(window);
         return window;
     }


     /* this method will not return until closing the window
      * starts a new application by given Window baseWindow
      * */
     public void run(Window baseWindow)
     {
    	 // nothing to do with GWT
     }
     



}
